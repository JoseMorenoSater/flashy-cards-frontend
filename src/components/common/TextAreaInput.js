/**
 *
 * TextAreaInput
 *
 */

import React from "react";
import PropTypes from "prop-types";

const TextAreaInput = ({
  divClassName,
  label,
  inputClassName,
  inputId,
  onBlur,
  defaultValue
}) => {
  let text;
  return (
    <div className={divClassName}>
      <label htmlFor="lessonName">{label}</label>
      <textarea
        defaultValue={defaultValue}
        ref={node => (text = node)}
        type="text"
        className={inputClassName}
        id={inputId}
        onBlur={() => onBlur(text.value)}
      />
    </div>
  );
};

TextAreaInput.propTypes = {
  divClassName: PropTypes.string,
  label: PropTypes.string,
  inputClassName: PropTypes.string,
  inputId: PropTypes.string,
  onBlur: PropTypes.func,
  defaultValue: PropTypes.string
};

export default TextAreaInput;
