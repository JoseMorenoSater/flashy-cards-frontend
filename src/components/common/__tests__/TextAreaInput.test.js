import React from "react";
import { create } from "react-test-renderer";
import TextAreaInput from "../TextAreaInput";

describe("TextAreaInput Component", () => {
  it("should render correctly", () => {
    const TextAreaInputComponent = create(<TextAreaInput />).toJSON();
    expect(TextAreaInputComponent).toMatchSnapshot();
  });
});
