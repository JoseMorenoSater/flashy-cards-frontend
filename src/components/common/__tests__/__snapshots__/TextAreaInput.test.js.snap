// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaInput Component should render correctly 1`] = `
<div
  className={undefined}
>
  <label
    htmlFor="lessonName"
  />
  <textarea
    className={undefined}
    defaultValue={undefined}
    id={undefined}
    onBlur={undefined}
    onChange={undefined}
    type="text"
    value={undefined}
  />
</div>
`;
